ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/heightPlanning/LookAheadCoMHeightTrajectoryGenerator.java

```diff
@@ -419,6 +422,16 @@ private void computeWaypoints(FramePoint3DReadOnly startCoMPosition,
       fourthMidpoint.setXY(fourthMidpointX, fourthMidpointY);
       endWaypoint.setXY(endWaypointX, endWaypointY);
 
+
+      // Handle the case where the feet are basically squared up when stepping up. In that case, the max height needs to be pinned by the "from" foot, as the
+      // way the phase variable works means we can approach what is meant to be swing on the "to" foot while still in the "from" foot.
+      if (Math.abs(transferToPosition.getX()) < 0.1 && endGroundHeight > startGroundHeight)
+      {
+         double blend = Math.abs(transferToPosition.getX()) / 0.1;
+         double minHeight = Math.min(startGroundHeight + extraToeOffHeight, endGroundHeight);
+         endGroundHeight = EuclidCoreTools.interpolate(minHeight, endGroundHeight, blend);
+      }
```

Comment:
This handles a weird edge case, where the feet are squaring up, which could result in it trying to shift the height to the step up height when the other foot is still on the ground.